dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkApplicationListener.java

```diff
@@ -84,7 +85,7 @@ else if (event.getType() == RequestEvent.Type.RESP_FILTERS_START) {
                         commitTransaction();
                     } catch (Exception e) {
                         rollbackTransaction();
-                        this.<RuntimeException>rethrow(e);
+                        throw new MappableException(e);
```

Comment:
I'm a little bit uncomfortable with using an exception class clearly being used only (?) internally by Jersey. Maybe we could use a second `catch` block before this one which catches [`HibernateException`](https://docs.jboss.org/hibernate/orm/4.3/javadocs/org/hibernate/HibernateException.html) (which inherits from RuntimeException) and re-throw it unchanged while we use [`Throwables#propagateIfPossible(java.lang.Throwable)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Throwables.html#propagateIfPossible(java.lang.Throwable)) in the catch-all block (which is currently throwing the MappableException).